source/iface/usm_basic_concept.rst

```diff
@@ -2,288 +2,317 @@
+and these functions may be used on device allocations if a device
+supports ``sycl::aspect::usm_device_allocations``.
+
+.. rubric:: Example
+
+See `usm-example-2`_.
+
+
+Host allocations
+----------------
+
+Host allocations allow devices to directly read and write host
+memory inside of a kernel.
+
+Host allocations must also be obtained using SYCL routines
+instead of system allocation routines. While a device may remotely
+read and write a host allocation, the allocation does not migrate
+to the device - it remains in host memory.
+
+.. warning::
+
+  Users should take care to properly synchronize access to
+  host allocations between host execution and kernels.
+
+The total size of host allocations will be limited by the amount
+of pinnable-memory on the host on most systems.
+
+Support for host allocations on a specific device can be queried
+through ``sycl::aspect::usm_host_allocations``.
+
+Support for atomic modification of host allocations on a specific
+device can be queried through ``sycl::aspect::usm_atomic_host_allocations``.
+
+
+Shared allocations
+------------------
+
+Shared allocations implicitly share data between the host and devices.
+
+Data may move to where it is being used without the programmer explicitly
+informing the runtime. It is up to the runtime and backends to make sure
+that a shared allocation is available where it is used.
+
+Shared allocations must also be obtained using SYCL allocation routines
+instead of the system allocator.
+
+The maximum size of a shared allocation on a specific
+device, and the total size of all shared allocations
+in a context, are implementation-defined.
+
+Support for shared allocations on a specific device can be
+queried through ``sycl::aspect::usm_shared_allocations``.
+
+.. warning::
+
+  Not all devices may support concurrent access of a shared allocation with the host.
+
+If a device does not support this, host execution and device
+code must take turns accessing the allocation, so the host
+must not access a shared allocation while a kernel is executing.
+
+.. warning::
+
+  Host access to a shared allocation which is also accessed by an executing
+  kernel on a device that does not support concurrent access results in
+  undefined behavior.
+
+  If a device does support concurrent access, both the host and and
+  the device may atomically modify the same data inside an allocation.
+
+Allocations, or pieces of allocations, are now free to migrate to different
+devices in the same context that also support this capability.
+Additionally, many devices that support concurrent access may support
+a working set of shared allocations larger than device memory.
+
+Whether a device supports concurrent access with atomic modification of
+shared allocations can be queried through the aspect
+``sycl::aspect::usm_atomic_shared_allocations``.
+
+.. note::
+
+  In the most capable systems, users do not need to use SYCL USM
```

Comment:
The information in this note should be moved to a new sub-section named "System allocations".  That sub-section should be after "Performance hints" because the performance hints apply to the USM Shared allocations.